lang/src/org/partiql/lang/compiler/PartiQLCompilerPipeline.kt

```diff
@@ -0,0 +1,103 @@
+package org.partiql.lang.compiler
+
+import com.amazon.ion.system.IonSystemBuilder
+import org.partiql.lang.domains.PartiqlAst
+import org.partiql.lang.domains.PartiqlPhysical
+import org.partiql.lang.errors.PartiQLException
+import org.partiql.lang.eval.PartiQLStatement
+import org.partiql.lang.planner.PartiQLPlanner
+import org.partiql.lang.planner.PartiQLPlannerBuilder
+import org.partiql.lang.syntax.Parser
+import org.partiql.lang.syntax.SqlParser
+
+/**
+ * [PartiQLCompilerPipeline] is the top-level class for embedded usage of PartiQL.
+ *
+ * Example usage:
+ * ```
+ * val pipeline = PartiQLCompilerPipeline.standard()
+ * val session = // session bindings
+ * val statement = pipeline.compile("-- some PartiQL query!")
+ * val result = statement.eval(session)
+ * when (result) {
+ *   is PartiQLResult.Value -> handle(result)  // Query Result
+ *   is PartiQLResult.Insert -> handle(result) // DML `Insert`
+ *   is PartiQLResult.Delete -> handle(result) // DML `Delete`
+ *   ...
+ * }
+ * ```
+ */
+class PartiQLCompilerPipeline private constructor(
+    private val parser: Parser,
+    private val planner: PartiQLPlanner,
+    private val compiler: PartiQLCompiler
+) {
+
+    companion object {
+
+        private val DEFAULT_ION = IonSystemBuilder.standard().build()
+
+        /**
+         *
+         */
+        @JvmStatic
+        fun standard() = PartiQLCompilerPipeline(
+            parser = SqlParser(DEFAULT_ION),
+            planner = PartiQLPlannerBuilder.standard().build(),
+            compiler = PartiQLCompilerBuilder.standard().build()
+        )
+
+        /**
+         * Builder utility for easy pipeline creation.
+         *
+         * Example usage:
+         * ```
+         *
+         * ```
+         */
```

Comment:
Just checking: do we intend to fill this in a subsequent PR? same applies to `standard()` documentation.